### PR TITLE
"more" view to show the A-Z links

### PR DESCRIPTION
**Pull request by drewvolz**
Show the A-Z urls in a grouped section list. 

- Each list row opens a webview.
- The JSON data originates [from here](https://wp.stolaf.edu/wp-json/site-data/sidebar/a-z).

Todo:
- [x] Type fixups
- [x] Camel case renamings
- [x] Search
- Proxy the endpoint on ccc?

---

📸 Screenshots

List | Detail
--|--
Simulator Screen Shot - iPhone 12 Pro - 2022-06-28 at 18 38 35 | detail

### CODE REVIEW

**Comment by hawkrives** [comment]
Merge conflicts gone wrong?

**Comment by drewvolz** [comment]
Strange things are happening with the tsc errors and counts. These are on `master` and therefore this branch...

<details>
<summary>diff</summary>

```diff
diff --git a/tsc-counts b/tsc-counts
index d0e8004fd..4731355bf 100644
--- a/tsc-counts
+++ b/tsc-counts
@@ -1,6 +1,6 @@
 21 TS2345
 18 TS2740
-17 TS2322
+18 TS2322
 5 TS7016
 4 TS2769
 4 TS2307
diff --git a/tsc-errors b/tsc-errors
index a327e5d98..2e0f80b75 100644
--- a/tsc-errors
+++ b/tsc-errors
@@ -18,7 +18,7 @@ source/lib/login.ts(57,4): error TS2769: No overload matches this call.
   Overload 1 of 2, '(input: RequestInfo, init?: RequestInit | undefined): Promise<Response>', gave the following error.
     Argument of type '{ method: string; body: FormData; credentials: "include"; cache: "no-store"; throwHttpErrors: boolean; }' is not assignable to parameter of type 'RequestInit'.
       Object literal may only specify known properties, and 'throwHttpErrors' does not exist in type 'RequestInit'.
-  Overload 2 of 2, '(input: RequestInfo | URL, init?: RequestInit | undefined): Promise<Response>', gave the following error.
+  Overload 2 of 2, '(input: RequestInfo, init?: RequestInit | undefined): Promise<Response>', gave the following error.
     Argument of type '{ method: string; body: FormData; credentials: "include"; cache: "no-store"; throwHttpErrors: boolean; }' is not assignable to parameter of type 'RequestInit'.
       Object literal may only specify known properties, and 'throwHttpErrors' does not exist in type 'RequestInit'.
 source/lib/login.ts(74,29): error TS2571: Object is of type 'unknown'.
@@ -39,6 +39,7 @@ source/navigation/track.ts(12,30): error TS2345: Argument of type 'NavigationRou
   Type 'NavigationRoute<ParamListBase, string>' is missing the following properties from type 'Readonly<{ key: string; index: number; routeNames: string[]; history?: unknown[] | undefined; routes: NavigationRoute<ParamListBase, string>[]; type: string; stale: false; }>': index, routeNames, routes, type, stale
 source/navigation/track.ts(14,15): error TS2339: Property 'routeName' does not exist on type 'NavigationRoute<ParamListBase, string>'.
 source/redux/init.ts(27,36): error TS2345: Argument of type 'ThunkAction<GetEnabledToolsAction>' is not assignable to parameter of type 'AnyAction'.
+  Property 'type' is missing in type 'ThunkAction<GetEnabledToolsAction>' but required in type 'AnyAction'.
 source/redux/parts/help.ts(61,22): error TS2322: Type '{ fetching: boolean; tools: never[]; lastFetchError: null; }' is not assignable to type 'State'.
   Types of property 'lastFetchError' are incompatible.
     Type 'null' is not assignable to type 'boolean | undefined'.
@@ -62,6 +63,7 @@ source/redux/parts/stoprint.ts(133,40): error TS2345: Argument of type 'string |
 source/views/building-hours/detail/schedule-row-android.tsx(37,22): error TS7031: Binding element 'children' implicitly has an 'any' type.
 source/views/building-hours/detail/schedule-row-android.tsx(37,32): error TS7031: Binding element 'style' implicitly has an 'any' type.
 source/views/building-hours/detail/toolbar-button.tsx(20,18): error TS2345: Argument of type 'ThunkAction<ToggleFavoriteAction>' is not assignable to parameter of type 'AnyAction'.
+  Property 'type' is missing in type 'ThunkAction<ToggleFavoriteAction>' but required in type 'AnyAction'.
 source/views/building-hours/lib/__tests__/get-detailed-building-status.test.ts(10,3): error TS2740: Type '{}' is missing the following properties from type 'BreakScheduleContainerType': fall, thanksgiving, christmasfest, winter, and 4 more.
 source/views/building-hours/lib/__tests__/get-detailed-building-status.test.ts(39,3): error TS2740: Type '{}' is missing the following properties from type 'BreakScheduleContainerType': fall, thanksgiving, christmasfest, winter, and 4 more.
 source/views/building-hours/lib/__tests__/get-detailed-building-status.test.ts(63,3): error TS2740: Type '{}' is missing the following properties from type 'BreakScheduleContainerType': fall, thanksgiving, christmasfest, winter, and 4 more.
@@ -97,6 +99,7 @@ source/views/settings/screens/debug/row.tsx(43,5): error TS2322: Type '((key: st
   Type 'null' is not assignable to type '(() => false | void) | undefined'.
 source/views/settings/screens/overview/app-icon.tsx(2,24): error TS7016: Could not find a declaration file for module '@hawkrives/react-native-alternate-icons'. './node_modules/@hawkrives/react-native-alternate-icons/index.js' implicitly has an 'any' type.
   Try `npm i --save-dev @types/hawkrives__react-native-alternate-icons` if it exists or add a new declaration (.d.ts) file containing `declare module '@hawkrives/react-native-alternate-icons';`
+source/views/settings/screens/overview/developer.tsx(16,46): error TS2322: Type '"info"' is not assignable to type 'Severity | undefined'.
 source/views/settings/screens/overview/login-credentials.tsx(140,13): error TS2345: Argument of type 'ThunkAction<LogInActions>' is not assignable to parameter of type 'AnyAction'.
 source/views/settings/screens/overview/login-credentials.tsx(146,12): error TS2345: Argument of type 'Promise<LogOutAction>' is not assignable to parameter of type 'AnyAction'.
   Property 'type' is missing in type 'Promise<LogOutAction>' but required in type 'AnyAction'.
@@ -111,6 +114,7 @@ source/views/settings/screens/overview/server-url.tsx(46,5): error TS2769: No ov
 source/views/sis/balances-acknowledgement.tsx(23,18): error TS2345: Argument of type 'Promise<SisAlertSeenAction>' is not assignable to parameter of type 'AnyAction'.
   Property 'type' is missing in type 'Promise<SisAlertSeenAction>' but required in type 'AnyAction'.
 source/views/sis/balances.tsx(197,13): error TS2345: Argument of type 'ThunkAction<LogInActions>' is not assignable to parameter of type 'AnyAction'.
+  Property 'type' is missing in type 'ThunkAction<LogInActions>' but required in type 'AnyAction'.
 source/views/sis/course-search/detail/index.tsx(126,5): error TS2322: Type 'Element[]' is not assignable to type 'string'.
 source/views/sis/course-search/detail/index.tsx(209,24): error TS7053: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{ Open: string; Closed: string; }'.
   No index signature with a parameter of type 'string' was found on type '{ Open: string; Closed: string; }'.
@@ -123,9 +127,13 @@ source/views/sis/course-search/list.tsx(104,4): error TS2322: Type '{ icon: stri
 source/views/sis/course-search/results.tsx(102,4): error TS2322: Type 'boolean | FilterType[]' is not assignable to type 'boolean'.
   Type 'FilterType[]' is not assignable to type 'boolean'.
 source/views/sis/course-search/results.tsx(122,31): error TS2345: Argument of type 'ThunkAction<UpdateRecentSearchesAction>' is not assignable to parameter of type 'AnyAction'.
+  Property 'type' is missing in type 'ThunkAction<UpdateRecentSearchesAction>' but required in type 'AnyAction'.
 source/views/sis/course-search/results.tsx(126,39): error TS2345: Argument of type 'ThunkAction<UpdateRecentFiltersAction>' is not assignable to parameter of type 'AnyAction'.
+  Property 'type' is missing in type 'ThunkAction<UpdateRecentFiltersAction>' but required in type 'AnyAction'.
 source/views/sis/course-search/search.tsx(205,18): error TS2345: Argument of type 'LoadCourseDataActionType' is not assignable to parameter of type 'AnyAction'.
+  Property 'type' is missing in type 'LoadCourseDataActionType' but required in type 'AnyAction'.
 source/views/sis/course-search/search.tsx(209,18): error TS2345: Argument of type 'UpdateCourseDataActionType' is not assignable to parameter of type 'AnyAction'.
+  Property 'type' is missing in type 'UpdateCourseDataActionType' but required in type 'AnyAction'.
 source/views/sis/course-search/search.tsx(216,4): error TS2322: Type '() => AnyAction' is not assignable to type '() => Promise<any>'.
   Type 'AnyAction' is missing the following properties from type 'Promise<any>': then, catch, finally, [Symbol.toStringTag]
 source/views/sis/course-search/search.tsx(219,4): error TS2322: Type '() => AnyAction' is not assignable to type '() => Promise<any>'.
@@ -133,9 +141,11 @@ source/views/sis/course-search/search.tsx(219,4): error TS2322: Type '() => AnyA
 source/views/sis/student-work/index.tsx(62,41): error TS2304: Cannot find name 'mixed'.
 source/views/stoprint/print-jobs.tsx(199,36): error TS2345: Argument of type 'ThunkAction<LogInActions>' is not assignable to parameter of type 'AnyAction'.
 source/views/stoprint/print-jobs.tsx(203,18): error TS2345: Argument of type 'ThunkAction<UpdatePrintJobsAction>' is not assignable to parameter of type 'AnyAction'.
+  Property 'type' is missing in type 'ThunkAction<UpdatePrintJobsAction>' but required in type 'AnyAction'.
 source/views/stoprint/print-jobs.tsx(211,4): error TS2322: Type '(username: any, password: any) => AnyAction' is not assignable to type '(username: string, password: string) => Promise<any>'.
   Type 'AnyAction' is not assignable to type 'Promise<any>'.
 source/views/stoprint/print-jobs.tsx(213,4): error TS2322: Type '() => AnyAction' is not assignable to type '() => Promise<any>'.
   Type 'AnyAction' is not assignable to type 'Promise<any>'.
 source/views/stoprint/printers.tsx(163,18): error TS2345: Argument of type 'ThunkAction<UpdateAllPrintersAction>' is not assignable to parameter of type 'AnyAction'.
+  Property 'type' is missing in type 'ThunkAction<UpdateAllPrintersAction>' but required in type 'AnyAction'.
 source/views/types.ts(5,8): error TS2307: Cannot find module 'react-navigation' or its corresponding type declarations.
```

</details>

**Comment by hawkrives** [comment]
I updated TSC on master yesterday (? Or over the weekend sometime)

**Comment by hawkrives** [comment]
@drewvolz shouldn't we use the ">" list row type, since we're opening in an in-app webview?

**Comment by drewvolz** [comment]
We could do that. Is that standard, though, to have them be blue and have a chevron?

**Comment by hawkrives** [comment]
No, I don't think they'd be blue at that point

**Comment by drewvolz** [comment]
This is ready for another review.

File: source/views/more/index.tsx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+}
+
+function MoreView(): JSX.Element {
+	let navigation = useNavigation()
+
+	let [query, setQuery] = React.useState('')
+	let searchQuery = useDebounce(query.toLowerCase(), 200)
+
+	let {
+		data: {az_nav: {menu_items: menuItems = []} = {}} = {},
```
Comment (hawkrives) [inline source/views/more/index.tsx]:
```suggestion
		data,
```

Then later

```ts 
let menuItems = data?.az_nav?.menu_items ?? []
```

**Review by hawkrives — APPROVED** [review]

File: source/views/more/index.tsx (hunk 1 above)
Comment (drewvolz) [reply source/views/more/index.tsx]: Great idea. This comment inspired me to break this up into two parts:
- `data = []` lives at #6206 
- `data?.az_nav?.menu_items` lives at https://github.com/frog-pond/ccc-server/pull/617

**Review by hawkrives — APPROVED** [review]

**Pull request merged**